x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -155,6 +155,30 @@
                                 RoleDescriptor.IndicesPrivileges.builder()
                                     .indices(".logs-endpoint.diagnostic.collection-*")
                                     .privileges("read").build(),
+                                // Fleet Server indices. Kibana read and write from these indices to manage Elastic Agents.
+                                // Kibana write to this indice to reassign agent policy or perform force unenroll
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices(".fleet-agents")
+                                    .privileges("read", "write").build(),
+                                // Kibana write to this indice to add action to an agent, upgrade, unenroll, ...
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices(".fleet-actions")
+                                    .privileges("read", "write").build(),
+                                // Kibana write to this indice new enrollment api key
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices(".fleet-enrollment-api-keys")
```

Comment:
TLDR: If there are Kibana authz checks that can be bypassed by directly reading the enrollment api keys index, I suggest we make it a restricted index. This might be the case for the other fleet indices as well.


----

Users that have read access to all indices, eg:
```
POST /_security/role/read_all_indices
{
  "indices": [
    {
      "names": [ "*" ],
      "privileges": ["all"]
    }
  ]
}
```
can access dot-named indices (which are implicitly hidden) using the index name or, in the case of wildcards, with a request option.

For indices that contain sensitive data (more than just configuration data or raw-metrics, which is a nuisance to expose but is not equivalent to bypassing any other checks we have in place) we have an extra protection level, named [restricted indices](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/index/RestrictedIndicesNames.java). The restricted indices concept is most likely to become superseded by system indices, but this is what we have and use currently.

Reply:
I was not aware of `restricted` indices @ruflin @blakerouse @aleksmaus do you think this will work for us. The mapping of this indices will be created with a regular user using an integration package.

Reply:
Was also not aware of restricted indices concept. @scunningham @jaymode One more option ... ?

Reply:
To maybe make it more clear what I mean, [restricted indices](https://www.elastic.co/guide/en/elasticsearch/reference/current/defining-roles.html#roles-indices-priv) confer addition protection by requiring explicit toggling of a flag when defining index permissions in a role, and relies on the expectation that roles in general don't have this flag toggled. Importantly, `.kibana*` is not a restricted index (but `.security*` is) which means that roles don't require the "hidden" flag toggled to grant access to it. Besides `kibana` there are other indices (think ML) that maybe should deserve more protection.

The fact that this is a new index and that it contains credentials prompted my suggestion.

But restricted indices is no silver bullet, we can just simply document non-fleet-admin users should not be granted privileges over these indices.

Reply:
@ruflin sorry I have not had a chance to communicate this, but restricted indices will be used for external system indices. @tvernum and I discussed this on the external system indices doc and then again last week. I'll begin working on implementing external system indices soon :)